src/main/java/de/tum/cit/aet/artemis/assessment/web/ResultResource.java

```diff
@@ -280,16 +282,60 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
+     * Sorting is applied by the specified sorted column and sorting order. If the provided sorted column is not valid for sorting (e.g., "taskNumber" or "errorCategory"),
+     * the sorting defaults to "count".
+     * <br>
+     * The response contains both the paginated feedback details and the total count of distinct results for the exercise.
      *
-     * @param exerciseId The ID of the exercise for which feedback details should be retrieved.
-     * @return A ResponseEntity containing a list of {@link FeedbackDetailDTO}s
+     * @param exerciseId       The ID of the exercise for which feedback details should be retrieved.
+     * @param page             The page number of the results to retrieve (0-indexed).
+     * @param pageSize         The number of feedback details per page.
+     * @param searchTerm       The search term used for filtering the results (optional).
+     * @param sortingOrder     The sorting order for the results (ASCENDING or DESCENDING).
+     * @param sortedColumn     The column by which the results should be sorted. If an invalid column (e.g., "taskNumber", "errorCategory") is provided, sorting defaults to
+     *                             "count".
+     * @param filterTasks      A list of task numbers to filter feedback details by the associated tasks (optional).
+     * @param filterTestCases  A list of test case names to filter feedback details by the associated test cases (optional).
+     * @param filterOccurrence A list of two values representing the minimum and maximum occurrence of feedback items to include (optional).
+     * @return A {@link ResponseEntity} containing a {@link FeedbackAnalysisResponseDTO}, which includes:
+     *         - {@link SearchResultPageDTO < FeedbackDetailDTO >} feedbackDetails: Paginated feedback details for the exercise.
+     *         - long totalItems: The total number of feedback items (used for pagination).
+     *         - int totalAmountOfTasks: The total number of tasks associated with the feedback.
+     *         - List<String> testCaseNames: A list of test case names included in the feedback.
      */
-    @GetMapping("exercises/{exerciseId}/feedback-details")
-    @EnforceAtLeastEditorInExercise
-    public ResponseEntity<List<FeedbackDetailDTO>> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
-        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
-        return ResponseEntity.ok(resultService.findAggregatedFeedbackByExerciseId(exerciseId));
+    @GetMapping("exercises/{exerciseId}/feedback-details-paged")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<FeedbackAnalysisResponseDTO> getFeedbackDetailsPaged(@PathVariable long exerciseId, @RequestParam int page, @RequestParam int pageSize,
+            @RequestParam(required = false) String searchTerm, @RequestParam String sortingOrder, @RequestParam String sortedColumn, @RequestParam List<String> filterTasks,
```

Comment:
```suggestion
            @RequestParam(required = false) String searchTerm, @RequestParam SortingOrder sortingOrder, @RequestParam String sortedColumn, @RequestParam List<String> filterTasks,
```
Why can't you use the enum directly here? I think this is also used in other endpoints e.g. via `PageableSearchDTO`